--- a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
@@ -581,6 +581,7 @@
 
 &i2c1{
 	status = "okay";
+	clock-frequency = <100000>;
 	ebc_pmic: pmic@68 {
 			status = "okay";
 			compatible = "ti,tps65185";
@@ -625,8 +626,6 @@
 					vcom: vcom {
 							regulator-name = "vcom";
 							/* voltage range is board-specific */
-							regulator-min-microvolt = <1170000>;
-                            regulator-max-microvolt = <1170000>;
 					};
 
 					vdrive: vdrive {
@@ -968,3 +967,18 @@
 &rga {
 	status = "okay";
 };
+
+&cru {
+	assigned-clocks = <&pmucru CLK_RTC_32K>,
+			  <&cru ACLK_TOP_HIGH>, <&cru ACLK_TOP_LOW>,
+			  <&cru PLL_GPLL>, <&pmucru PLL_PPLL>,
+			  <&cru ACLK_RGA_PRE>, <&cru CPLL_250M>,
+			  <&cru HCLK_RGA_PRE>, <&cru HCLK_EBC>,
+			  <&cru HCLK_JENC>, <&cru CPLL_333M>;
+	assigned-clock-rates = <32768>,
+			       <500000000>, <400000000>,
+			       <1200000000>, <200000000>,
+			       <300000000>, <250000000>,
+			       <300000000>, <300000000>,
+			       <300000000>, <33333334>;
+	assigned-clock-parents = <&pmucru CLK_RTC32K_FRAC>;
+};
